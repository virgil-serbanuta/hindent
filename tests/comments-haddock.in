-- Haddock comments

-- | Module comment.
module X where

-- | Main doc.
main :: IO ()
main = return ()

data X
  = X -- ^ X is for xylophone.
  | Y -- ^ Y is for why did I eat that pizza.

data X = X
  { field1 :: Int -- ^ Field1 is the first field.
  , field11 :: Char
    -- ^ This field comment is on its own line.
  , field2 :: Int -- ^ Field2 is the second field.
  , field3 :: Char -- ^ This is a long comment which starts next to
    -- the field but continues onto the next line, it aligns exactly
    -- with the field name.
  , field4 :: Char
    -- ^ This is a long comment which starts on the following line
    -- from from the field, lines continue at the sme column.
  }
