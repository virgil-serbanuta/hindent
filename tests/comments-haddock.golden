-- Haddock comments
-- | Module comment.
module X where

-- | Main doc.
main :: IO ()
main = return ()

data X
    = X -- ^ X is for xylophone.
    | Y -- ^ Y is for why did I eat that pizza.

data X = X
    { field1  :: Int
    , field11 :: Char
    , field2  :: Int
    , field3  :: Char
    , field4  :: Char
    }
